### PR TITLE
Add json export

### PR DESCRIPTION
**Pull request by dood-**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #64

### CODE REVIEW

File: src/VarDumper.php [hunk 1]
```diff
@@ -295,6 +321,37 @@ private function exportInternal($variable, bool $format, int $level): string
...
 
+    /**
+     * @param mixed $var
+     * @param bool $format
+     * @param int $depth
+     * @param int $level
+     * @return mixed
+     * @throws ReflectionException
+     * @psalm-param mixed $var
+     */
```
Comment (samdark) [inline src/VarDumper.php:332]: Looks like it duplicates signature of the method itself.

```suggestion
```

File: README.md [hunk 2]
```diff
@@ -85,6 +85,19 @@ That is especially useful if an object contains callbacks that should get info f
...
+```php
+$string = \Yiisoft\VarDumper\VarDumper::create($variable)->asJson();
+```
+
+It is similar to `json_encode()` but uses short array syntax, handles closures, and serializes objects.
+
+In the above `asJson()` will give you nicely formatted code. You can remove formatting by passing `false` as the first
+`$format` argument.
+
+`$depth` argument allows you to set maximum recursion depth
```
Comment (samdark) [inline README.md:99]:
```suggestion
`$depth` argument allows you to set maximum recursion depth.
```

**Review by samdark — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 ## 1.2.1 under development
 
-- no changes in this release.
+- Eng #64: Add `asJson()` method to `\Yiisoft\VarDumper\VarDumper` (dood-)
```
Comment (samdark) [inline CHANGELOG.md:6]:
```suggestion
- Enh #64: Add `asJson()` method to `\Yiisoft\VarDumper\VarDumper` (dood-)
```

**Comment by samdark** [comment]
Thanks!

**Comment by xepozz** [comment]
It doesn't work well with Json.

**Comment by samdark** [comment]
What do you mean?

**Comment by xepozz** [comment]
Objects dump not as json structure. Infinite array is not an array.

**Comment by xepozz** [comment]
https://github.com/yiisoft/var-dumper/pull/66

**Pull request merged**